packages/jitm/src/class-post-connection-jitm.php

```diff
@@ -465,11 +465,9 @@ public function get_messages( $message_path, $query, $full_jp_logo_exists ) {
 				)
 			);
 
-			$normalized_site_url = \Jetpack::build_raw_urls( get_home_url() );
-
 			$url_params = array(
 				'source' => "jitm-$envelope->id",
-				'site'   => $normalized_site_url,
+				'site'   => jetpack_get_site_suffix(),
```

Comment:
This is okay for now, but the JITM package shouldn't need the Jetpack plugin itself (as it does still right now).

This leads me to thinking the global function needs to be in a package, instead of the global functions for the plugin.

The use case for the function would be for Calypso (I think inclusive of cloud.jetpack.com) for a connected site, so any connected site could utilize it.

What would you think about it being part of the Connection package instead?

Reply:
I'm not sure I know enough about Jetpack's packages and their availability to have an opinion. Should it live in `Utils` or  somewhere else?

Reply:
@jeherve Flagging this for you since you suggested the global functions file originally. :)

Reply:
Chatting in Slack, probably the Status package.

Within Jetpack, would need ensure a `use` line is present for the status package, then call in the class.

Here's an example from WordAds: 
* use: https://github.com/Automattic/jetpack/blob/8ab0362810c20f6dec9f269d0d4fa3aa3dc82f0e/modules/wordads/php/class-wordads-api.php#L9

* actual usage: https://github.com/Automattic/jetpack/blob/8ab0362810c20f6dec9f269d0d4fa3aa3dc82f0e/modules/wordads/php/class-wordads-api.php#L34

With tests in the packages/status/tests dir.

Reply:
I've now moved the new function to the new Status package.

Should we also port the special case of WordPress.com that was in use in some of the past usages?

```diff
diff --git a/packages/status/src/class-status.php b/packages/status/src/class-status.php
index 8ee782c06..455f425d1 100644
--- a/packages/status/src/class-status.php
+++ b/packages/status/src/class-status.php
@@ -279,6 +279,14 @@ class Status {
 	 * @return string
 	 */
 	public function get_site_suffix( $url = '' ) {
+		// On WordPress.com, site suffixes are a bit different.
+		if (
+			class_exists( 'WPCOM_Masterbar' )
+			&& method_exists( 'WPCOM_Masterbar', 'get_calypso_site_slug' )
+		) {
+			$site_slug = \WPCOM_Masterbar::get_calypso_site_slug( get_current_blog_id() );
+		}
+
 		if ( empty( $url ) ) {
 			$url = \home_url();
 		}

```

Reply:
Yes, I think we should. I assumed `Jetpack::build_raw_urls()` would always be available on dotcom, but that doesn't look like that's the case

Reply:
Added in c8d9b3b